akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -768,18 +768,41 @@ object Source {
     })
 
   /**
-   * Combines several sources with fan-in strategy like [[Merge]] or [[Concat]] into a single [[Source]] with a materialized value.
+   * Combines several sources with fan-in strategy like [[Merge]] or [[Concat]] into a single [[Source]].
+   */
+  def combine[T, U, M](sources: immutable.Seq[Graph[SourceShape[T], M]])(
+      fanInGraph: Graph[UniformFanInShape[T, U], NotUsed]): Source[U, immutable.Seq[M]] =
```

Comment:
I don't remember any rationale but I'd guess the same as you, to be able to do `Merge(_)` and `Concat(_)`, kinda superfluous because you know the size, but OTOH possible to get wrong if it is not fed to the strategy but repeated by user.

Reply:
@He-Pin sorry for the extra work, but could you change back to the old `strategy: Int => ` so that we have a consistent api for these and what already exists.